### DISCUSSION TITLE
subscriptionExchange and SSR

### DISCUSSION THREAD

**Discussion by aldis-ameriks** [urql-graphql/urql#1087] [Q&A] [answered]
What's the optimal way to handle `subscriptionExchange` and `ssr`?

I'm now doing the following, which seems to work. But made me wonder what's the recommended way and/or how others are handling it?

```js
  if (!isServerSide) {
    const subscriptionClient = new SubscriptionClient(config.apiGraphqlWsEndpoint, { reconnect: true });

    exchanges.push(
      subscriptionExchange({
        forwardSubscription(operation) {
          return subscriptionClient.request(operation);
        },
      })
    );
  }
```

**Comment by kitten** [comment, marked as answer]
I'm not quite sure what the question here is. By default `useSubscription` shouldn't be taken into account for SSR, meaning that no subscription will actually be executed for runs using `react-ssr-prepass` for instance. So it's safe to either add a `subscriptionExchange` or leave it out for SSR.

**Reply by aldis-ameriks** [reply]
Based on types `subscriptionExchange` cannot be initialized without defining `forwardSubscription`.
However, when doing SSR, I want to skip the subscriptions client.
To work around that, I'm simply skipping the whole exchange altogether when doing SSR.
That's what the above example achieves.

But this results in warnings being thrown server side.
```
No exchange has handled operations of type "subscription". Check whether you've added an exchange responsible for these operations.
```

So this made me wonder if there's a better way of setting up the `subscriptionExchange`, that sets up the subscriptions client on browser, skips it in node, but also avoids the warnings, that occur when doing the server side rendering.

**Reply by aldis-ameriks** [reply]
After updating `urql` to `1.10.3`, the warnings are no longer being logged when skipping `subscriptionExchange` on node.
